ministryofjustice/laa-apply-for-criminal-legal-aid
# Change date type, remove evidence array erroneously added

## Description of change
Form was submitting fine, but on revisiting the page after submission the date was not repopulating the fields and blowing up.
Fixed data type of date_job_lost on people table - was string, needed to be date.

Also removes expeceted_evidence array that somehow crept into the schema on the previous implementation PR.

## Link to relevant ticket
https://ministryofjustice.sentry.io/issues/4598186677/?alert_rule_id=14332616&alert_timestamp=1699005354001&alert_type=email&environment=staging&notification_uuid=8bd06e84-e4f4-4a26-abc8-1cb87f78eda0&project=6587263&referrer=alert_email

## Notes for reviewer
Bug Hibo picked up on QA'ing the did_client_lose_job_being_in_custody - currently not working on staging. This can be seen by:
Starting a new application on staging
Add client details and save and continue
Navigate to /steps/income/did_client_lose_job_being_in_custody
Select yes, add in date
Save and continue
Click on the application you just created
Navigate to steps/income/did_client_lose_job_being_in_custody - this will blow up 

## How to manually test the feature
On local:
Start a new application
Add client details and save and continue
Navigate to /steps/income/did_client_lose_job_being_in_custody
Select yes, add in date
Save and continue
Click on the application you just created
Navigate to steps/income/did_client_lose_job_being_in_custody - Date should now be autopopulated with the date initially entered
